### PR TITLE
Ease embedding into host CMake projects during build

### PR DESCRIPTION
**Pull request by aradi**
- Allows overwriting of cache variables through normal variables
  in host project.
- Export library name with namespace as Mbd::Mbd

NOTE: The first point could be much easier realized by `cmake_minimum_required(3.13)`, if this is an option.

### CODE REVIEW

**Comment by aradi** [comment]
OK, it is ready to go now (from my side). I am not sure, what causes codecov-drop, as I did not touch anything related to this (I guess).

**Comment by jhrmnn** [comment]
The Travis CI has Cmake 3.12, so I'll keep the version check.

The Codecov report shows zero change now, I think the first one was bad because of the failed Travis CI.

So, I had a lookk at how you use this change in DFTB+, and I've read some Cmake documentation, but I still don't understand how *exactly* does this work. Could you perhaps point me to some online document that explains this construct? Or did you figure this out on your own?

**Comment by aradi** [comment]
As with CMake, no proper documentation, just man-pages, blogs, etc... I've spent the last four weeks with searching around for various issues/strategies...

The are actually two changes above:

- The policy setting. This allows me to define the cache variables of a sub-project as normal variables in the host one without exposing them as cache variables to the user. This is described [here](https://cmake.org/cmake/help/v3.16/policy/CMP0077.html). We will change now to CMake 3.16 and the policy setting in your CMakeLists.txt will allow me to use the standard policy of 3.16 despite the lower CMake version requirement of your project. (By the way, Travis-CI works fine with CMake 3.16 if you choose "focal" as distribution.)

- Actually, the namespacing in the export is more a suggestion as it does not affect projects which invoke libMbd directly as subprojects. But this is current CMake practice, and  is partially described [here](https://cmake.org/cmake/help/v3.6/manual/cmake-packages.7.html), or you can also have a look at any modern Find* module in CMake, they always export namespaced targets.  But it does not affect us directly, I can remove that change from the PR, if you do not want to apply it yet.

The only "magic" on our side is, that although we use your targets directly (without the namespace), I still prefer the usage of the namespaced form of the ``Mbd`` target. This way, I would not have to change much, if I ever change our build system and treat libMbd as external dependency (and search for it via its CMake export file which exports namespaced targets). Therefore, currently I just create the target ``Mbd::Mbd`` when incorporating libMbd-s build system and use that...

**Comment by jhrmnn** [comment]
> * The policy setting.

Yes, this I understand. I believe this enables you to add the root directory of Libmbd in DFTB+ rather than the `src` subdirectory.

Right, I could use the newer distro, I just thought that the default in Travis is a reasonable target to support.

> * Actually, the namespacing in the export is more a suggestion as it does not affect projects which invoke libMbd directly as subprojects. But this is current CMake practice, and  is partially described [here](https://cmake.org/cmake/help/v3.6/manual/cmake-packages.7.html), or you can also have a look at any modern Find* module in CMake, they always export namespaced targets.  But it does not affect us directly, I can remove that change from the PR, if you do not want to apply it yet.

The namespace I understand, and I agree is reasonable. What I don't understand is creating the interface library `Mbd`, linking `mbd` into it, and adding it to the export target. What's the purpose of that? Couldn't that be left out, and use `Mbd::mbd` in DFTB+ directly rather than `Mbd::Mbd`?

Thanks for taking the time to explain this!

**Comment by aradi** [comment]
> The namespace I understand, and I agree is reasonable. What I don't understand is creating the interface library `Mbd`, linking `mbd` into it, and adding it to the export target. What's the purpose of that? Couldn't that be left out, and use `Mbd::mbd` in DFTB+ directly rather than `Mbd::Mbd`?

Sure, I could use `Mbd::mbd` indeed. However, it seems, that many modern CMake finders export a target with the same name as the package itself, which would be `Mbd::Mbd` in your case. (Also case convention seems to be CamelCase for packages.) So I myself now sort of follow the convention, that packages export an abstract target with the same name as the package (and which is not directly mapped to a file name). That target offers the full functionality of the package. (If you then ever decide to split your functionality in let's say targets/sub-libraries `mbd` and `mbdext`, I'd be still able to use the  `Mbd::Mbd` target in our project, as long as you define the abstract `Mbd` target to depend on the two actual physical libraries `mbd` and `mbdext`).

**Comment by aradi** [comment]
One more comment: Invoking your project by only adding the `src` subdirectory turned out to be a bad practice. Important variables (e.g. `PROJECT_VERSION`) are not set correctly. And anyway, IMO, host projects should have no knowledge about or rely on the actual directory structure of a sub-project.

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,9 +1,6 @@
 cmake_minimum_required(VERSION 3.1)
 cmake_policy(SET CMP0042 NEW)
-
-# Allow defining cache variables as normal variables in a host project.
-if(CMAKE_MAJOR_VERSION GREATER 3
-    OR (CMAKE_MAJOR_VERSION EQUAL 3 AND CMAKE_MINOR_VERSION GREATER_EQUAL 13))
+if(NOT CMAKE_VERSION VERSION_LESS 3.13)
```
Comment (aradi) [inline CMakeLists.txt:3]: Cool! That's much nicer!

**Comment by jhrmnn** [comment]
> Sure, I could use Mbd::mbd indeed. However, it seems, that many modern CMake finders export a target with the same name as the package itself

I see. Ok, let's follow that convention.

>  Invoking your project by only adding the src subdirectory turned out to be a bad practice.

Agreed. I just didn't know how to do it better. Thanks for figuring it out!

**Pull request merged**